feat: amplifier docs in one place (#841)
axelarnetwork/axelar-docs: src/pages/validator/amplifier/integrator-onboarding.mdx

## Set up your CosmWasm devnet environment

1. Install the `v0.35.5` axelard binary. You can either:
    * Download the pre-built [`axelard v0.35.5`](https://github.com/axelarnetwork/axelar-core/releases/tag/v0.35.5) binary from GitHub for your device. You will also need to download 
    [`wasmvm v1.3.1`](https://github.com/CosmWasm/wasmvm/releases/tag/v1.3.1) `libwasmvm.x86_64.so` (for Linux) or `libwasmvm.dylib` (for Mac) and in either case add the parent folder to `LD_LIBRARY_PATH`, OR: 
    * Build `axelard` binary directly from the `axelar-core`: 
        * Clone the [repo](https://github.com/axelarnetwork/axelar-core)
        * Checkout the `v0.35.5` release tag
        * Run `make build`
1. Generate an account by running `axelard keys add wallet --keyring-backend test`.
1. Fund your devnet account with `uamplifier` tokens:
    - Join the [Axelar Discord](https://discord.com/invite/axelar).
    - Get the Developer Role.
    - Go to the [**#faucet**](https://discord.com/channels/770814806105128977/1002423218772136056/1217885883152334918) channel and submit a request with your account to get 100 test tokens: `!faucet devnet-amplifier [your axelar wallet address]`

> 1. Query your account balance by running `axelard q bank balances axelar1suw3fsckht00c5srgtr8p6lhn58e9lxq9yf5th —-node [RPC]`.

Reviewer: ```suggestion
1. Query your account balance by running `axelard q bank balances [your axelar wallet] —-node [RPC]`.
```

1. Deploy your CosmWasm contract.

## Run `ampd` / `tofnd`

See the [Become an Amplifier Verifier (Worker)](/validator/amplifier/verifier-onboarding) for detailed instructions on how to run ampd and tofnd on your machine.

## Onboard a chain to Amplifier

> 1. Build all of the deployed contracts, such as the router and service registry, from [`devnet-amplifier.json`](https://github.com/axelarnetwork/axelar-contract-deployments/blob/main/axelar-chains-config/info/devnet-amplifier.json) using the [Cosmwasm `rust-optimizer`](https://github.com/CosmWasm/optimizer). From the root of the repo, run:

Reviewer: this is incorrect, see my previous comment on this

You need to build contracts from the `axelar-amplifier` [repo](https://github.com/axelarnetwork/axelar-amplifier).  They need to checkout `v0.1.0` release to build. The deployments repo only hosts the config of *deployed* contracts, not the source code.

As CJ mentioned, you don't need to build all amplifier contracts to onboard your chain.

> 1. Build all of the deployed contracts, such as the router and service registry, from [`devnet-amplifier.json`](https://github.com/axelarnetwork/axelar-contract-deployments/blob/main/axelar-chains-config/info/devnet-amplifier.json) using the [Cosmwasm `rust-optimizer`](https://github.com/CosmWasm/optimizer). From the root of the repo, run:

Reviewer: ```suggestion
1. You can instantiate the gateway and voting verifier contracts with code id from the existing [deployment](https://github.com/axelarnetwork/axelar-contract-deployments/blob/main/axelar-chains-config/info/devnet-amplifier.json#L409).
1. Otherwise, you can build the amplifier contracts yourself from the [axelar-amplifier repo](https://github.com/axelarnetwork/axelar-amplifier), or build your own if you have a custom contract, using the [Cosmwasm `rust-optimizer`](https://github.com/CosmWasm/optimizer).
```

    ```bash
    docker run --rm -v "$(pwd)":/code \
    --mount type=volume,source="$(basename "$(pwd)")_cache",target=/target \
    --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
    cosmwasm/optimizer:0.15.0
    ```

> 1. Deploy the verifier:

Reviewer: ```suggestion
1. Deploy the voting verifier. You can also instantiate the [existing](https://github.com/axelarnetwork/axelar-contract-deployments/blob/main/axelar-chains-config/info/devnet-amplifier.json#L370) voting verifier contract which has code id `6` instead of deploying a new one:
```

>     axelard tx wasm instantiate 9 '{"service_registry_address":"axelar1hrpna9v7vs3stzyd4z3xf00676kf78zpe2u5ksvljswn2vnjp3ystlgl4x","service_name":"validators","source_gateway_address":"0xCa85f85C72df5f8428a440887CA7c449D94e0D0c","confirmation_height":1,"block_expiry":10,"source_chain":"test","rewards_address":"axelar1wkwy0xh89ksdgj9hr347dyd2dw7zesmtrue6kfzyml4vdtz6e5ws2pvc5e","voting_threshold":["9","10"]}' --from wallet --chain-id devnet-amplifier --gas auto --gas-adjustment 1.5 --gas-prices 0.007uamplifier --node http://devnet-amplifier.axelar.dev:26657 --label test-voting-verifier --no-admin

Reviewer: Instantiate step should have a separate number so it's easy to read

>     axelard tx wasm instantiate 9 '{"service_registry_address":"axelar1hrpna9v7vs3stzyd4z3xf00676kf78zpe2u5ksvljswn2vnjp3ystlgl4x","service_name":"validators","source_gateway_address":"0xCa85f85C72df5f8428a440887CA7c449D94e0D0c","confirmation_height":1,"block_expiry":10,"source_chain":"test","rewards_address":"axelar1wkwy0xh89ksdgj9hr347dyd2dw7zesmtrue6kfzyml4vdtz6e5ws2pvc5e","voting_threshold":["9","10"]}' --from wallet --chain-id devnet-amplifier --gas auto --gas-adjustment 1.5 --gas-prices 0.007uamplifier --node http://devnet-amplifier.axelar.dev:26657 --label test-voting-verifier --no-admin

Reviewer: ```suggestion
    axelard tx wasm instantiate [code id] '{"service_registry_address":"axelar1hrpna9v7vs3stzyd4z3xf00676kf78zpe2u5ksvljswn2vnjp3ystlgl4x","service_name":"validators","source_gateway_address":"0xCa85f85C72df5f8428a440887CA7c449D94e0D0c","confirmation_height":1,"block_expiry":10,"source_chain":"test","rewards_address":"axelar1wkwy0xh89ksdgj9hr347dyd2dw7zesmtrue6kfzyml4vdtz6e5ws2pvc5e","voting_threshold":["9","10"]}' --from wallet --chain-id devnet-amplifier --gas auto --gas-adjustment 1.5 --gas-prices 0.007uamplifier --node http://devnet-amplifier.axelar.dev:26657 --label test-voting-verifier --no-admin
```
Stub the code id so the user replaces it with the actual value

    ```

> 1. Deploy the gateway:

Reviewer: ```suggestion
1. Deploy the gateway contract. If you have customized/built your own gateway, then you can deploy that. You can also instantiate the [existing](https://github.com/axelarnetwork/axelar-contract-deployments/blob/main/axelar-chains-config/info/devnet-amplifier.json#L409) gateway contract which has code id `7` instead of deploying a new one:
```
some folks have their own custom gateway and shouldn't deploy ours. otherwise they can instantiate directly instead of building their own

    ```

> 1. Deploy the prover:

Reviewer: ```suggestion
1. Deploy the multisig prover. You can also instantiate the [existing](https://github.com/axelarnetwork/axelar-contract-deployments/blob/main/axelar-chains-config/info/devnet-amplifier.json#L434) multisig prover contract which has code id `8` instead of deploying a new one:

```

    axelard tx wasm instantiate 11 '{"admin_address": "axelar1zlr7e5qf3sz7yf890rkh9tcnu87234k6k7ytd9","gateway_address":"axelar1rcmvfsn77pd6m04ctqj3wcu66pvrw9p265cdl72w4zarfup2rv7qpryg7d","multisig_address":"axelar1ufs3tlq4umljk0qfe8k5ya0x6hpavn897u2cnf9k0en9jr7qarqqa9263g","service_registry_address":"axelar1hrpna9v7vs3stzyd4z3xf00676kf78zpe2u5ksvljswn2vnjp3ystlgl4x","voting_verifier_address":"axelar1x3960tw9cml6xsqtvzt4gmw3scauaxdd83rhs9dmlpjfjf9z9s7qjx8g0j","destination_chain_id": "43113","signing_threshold": ["4","5"],"service_name": "validators","chain_name":"test","worker_set_diff_threshold": 1,"encoder": "abi","key_type": "ecdsa"}' --from wallet --chain-id devnet-amplifier --gas auto --gas-adjustment 1.5 --gas-prices 0.007uamplifier --node http://devnet-amplifier.axelar.dev:26657 --label test-prover --no-admin
    # note the contract address in the output

>     Enter keyring passphrase:
>     gas estimate: 365482

Reviewer: ```suggestion
```

    ```

> 1. Send the chain’s gateway and prover addresses to the Interop Labs team by filling out the [Axelar Express Integration Form](https://docs.google.com/forms/d/e/1FAIpQLSchD7P1WfdSCQfaZAoqX7DyqJOqYKxXle47yrueTbOgkKQDiQ/viewform). The team will register the chain and gateway with the router and authorize the prover.

Reviewer: ```suggestion
1. Send the chain’s gateway and multisig prover contract addresses to the Interop Labs team by filling out the [Axelar Amplifier Integration Form](https://docs.google.com/forms/d/e/1FAIpQLSchD7P1WfdSCQfaZAoqX7DyqJOqYKxXle47yrueTbOgkKQDiQ/viewform). The team will register the chain and gateway with the router and authorize the prover.
```
not express
